src/som/primitives/TimerPrim.java

```diff
@@ -0,0 +1,47 @@
+package som.primitives;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+import com.oracle.truffle.api.dsl.GenerateNodeFactory;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.source.SourceSection;
+
+import som.interpreter.actors.Actor;
+import som.interpreter.actors.SPromise;
+import som.interpreter.actors.SPromise.SResolver;
+import som.interpreter.actors.WrapReferenceNode;
+import som.interpreter.actors.WrapReferenceNodeGen;
+import som.interpreter.nodes.nary.BinaryComplexOperation;
+
+@GenerateNodeFactory
+@Primitive(primitive = "actorResolveProm:after:")
+public abstract class TimerPrim extends BinaryComplexOperation{
+  protected TimerPrim(final BinaryComplexOperation node) { super(node); }
+  protected TimerPrim(final boolean eagWrap, final SourceSection source) { super(eagWrap, source); }
+
+  @Child protected WrapReferenceNode wrapper = WrapReferenceNodeGen.create();
+
+  @Specialization
+  public final Object doResolveAfter(final VirtualFrame frame, final SResolver resolver, final long timeout) {
+    Timer t = new Timer();
```

Comment:
just seeing this one. I think, I am going to change that before merging to be a field of the node. Or perhaps even a static field. We don't want a separate timer thread for each time out :)